packages/launchdarkly-adapter/modules/adapter/adapter.js

```diff
@@ -1,25 +1,56 @@
+// @flow
+
 import { initialize } from 'ldclient-js';
 import camelCase from 'lodash.camelcase';
 
-const adapterState = {
+type FlagValue = boolean | string;
+type FlagName = string;
+type User = {
+  key: string,
+};
+type Client = {
+  identify: (nextUser: User) => void,
+  on: (state: string, () => void) => void,
+  on: (state: string, (flagName: FlagName) => void) => void,
+  allFlags: () => Flags | null,
+};
+type Flag = [FlagName, FlagValue];
+type Flags = { [FlagName]: FlagValue };
+type OnFlagsStateChangeCallback = Flags => void;
+type OnStatusStateChangeCallback = ({ [string]: boolean }) => void;
```

Comment:
Have you considered putting the types into its own file?

Reply:
Yeap. I wanted to take a step at a time and see what the overlap actually is between packages. I don't want to have a `package/types` as that feels kind of strange when running in independent mode in lerna. WDYT?

Reply:
Hmm, depends what you want to do with the types, but no I would just simply have e.g. a `types.js` file in each package.

Reply:
But of course it would make sense if you are going to reuse those types. It's also perfectly fine to have types within the module itself if they are scoped there.

Reply:
Yeah. I planned on starting - during the refactoring - to have them in each module. Then to move them to a package level `types.js`. The stop, no mono-repo wide types. As that feels like shared libraries in microservice land.

Reply:
a58d575